core/trino-main/src/main/java/io/trino/execution/SqlTaskManager.java

```diff
@@ -366,6 +371,18 @@ public VersionedDynamicFilterDomains acknowledgeAndGetNewDynamicFilterDomains(Ta
 
     @Override
     public TaskInfo updateTask(Session session, TaskId taskId, Optional<PlanFragment> fragment, List<TaskSource> sources, OutputBuffers outputBuffers, OptionalInt totalPartitions)
+    {
+        try {
+            return embedVersion.embedVersion(() -> doUpdateTask(session, taskId, fragment, sources, outputBuffers, totalPartitions)).call();
```

Comment:
Can we maybe instead wire version embedder into jetty. So it is applied to all jetty threads automatically?

Reply:
that would work for some cases, but not for others, when things are running from thread pool
we could need to give up on "exactly once" goal, which may be OK, but will look a bit like a mess

i'm open to future improvements

Reply:
I would not care too much about "exactly once". But we can leave it as a potential improvement. Let's chat offline.

Reply:
discussed offline. we decided to merge as is and iterate